tests/filechecker.py

```diff
@@ -0,0 +1,103 @@
+import sys
+import os
```

Comment:
I am not sure why this is an external tool. The script that runs and checks can be the same file. This will save a bunch of logic and simplify things.